src/main/java/org/broadinstitute/consent/http/resources/UserResource.java

```diff
@@ -175,11 +175,11 @@ public Response update(@Auth AuthUser authUser, @Context UriInfo info, @PathPara
         try {
             UserUpdateFields userUpdateFields = gson.fromJson(json, UserUpdateFields.class);
             // Ensure that we have a real user with this ID, fail if we do not.
-            userService.findUserById(userId);
+            User user = userService.findUserById(userId);
             URI uri = info.getRequestUriBuilder().path("{id}").build(userId);
-            User user = userService.updateUserFieldsById(userUpdateFields, userId);
+            User updatedUser = userService.updateUserFieldsById(user, userUpdateFields, userId);
             Gson gson = new Gson();
-            JsonObject jsonUser = userService.findUserWithPropertiesByIdAsJsonObject(authUser, user.getUserId());
+            JsonObject jsonUser = userService.findUserWithPropertiesByIdAsJsonObject(authUser, updatedUser.getUserId());
             return Response.ok(uri).entity(gson.toJson(jsonUser)).build();
         } catch (Exception e) {
```

Comment:
I would suggest a completely new endpoint here. This one requires `@PathParam("id") Integer userId`. The new one would handle the same path without the ID param. Then you can just get the user from `@Auth AuthUser authUser` instead of looking them up for use in the update method.